src/veh_type.h

```diff
@@ -140,118 +140,85 @@ struct transform_terrain_data {
     time_duration post_field_age = 0_turns;
 };
 
+const std::vector<std::pair<std::string, std::string>> vpart_variants = {
```

Comment:
Is this the only thing that needs to be edited to add new part variants? If that's the case then the limitation isn't a big problem, I can add more to this list. Should this be pulling from a JSON list though? Or would that be better as another pr? It doesn't seem like the sort of thing that needs to be encoded in CPP to me

Reply:
This is the master list of names.  It could be moved to JSON, though I'd rather not.

Any changes to the names would need to be made here and in the doc file.  So it is pretty easy to update, at least.